lms/djangoapps/discussion/static/discussion/js/discussion_board_factory.js

```diff
@@ -0,0 +1,52 @@
+(function(define) {
+    'use strict';
+
+    define(
+        [
+            'jquery',
+            'backbone',
+            'discussion/js/discussion_router',
+            'common/js/discussion/views/new_post_view'
+        ],
+        function($, Backbone, DiscussionRouter, NewPostView) {
+            return function(options) {
+                var userInfo = options.user_info,
+                    sortPreference = options.sort_preference,
+                    threads = options.threads,
+                    threadPages = options.thread_pages,
+                    contentInfo = options.content_info,
+                    user = new window.DiscussionUser(userInfo),
+                    discussion,
+                    courseSettings,
+                    newPostView,
+                    router;
+                // TODO: Perhaps eliminate usage of global variables when possible
+                window.DiscussionUtil.loadRoles(options.roles);
+                window.$$course_id = options.courseId;
```

Comment:
Is there some special significance for why this `course_id` has to be preceded by two `$`? If there is, maybe it should be documented in a comment?